tests/admin_filters/tests.py

```diff
@@ -1099,3 +1105,108 @@ def test_list_filter_queryset_filtered_by_default(self):
         changelist = modeladmin.get_changelist_instance(request)
         changelist.get_results(request)
         self.assertEqual(changelist.full_result_count, 4)
+
+    def test_should_filter_by_date_hierarchy(self):
+        modeladmin = BookAdminWithDateHierarchy(Book, site)
+
+        def _test_date_hierarchy(query, expected_from_date, expected_to_date):
```

Comment:
Query: do we prefer these helper functions to be named like `assertDateHierarchy`?

Reply:
I followed the convntion used [a few tests up](https://github.com/hakib/django/blob/4e55f98c2bf3adf4bdcafa248349abe9f8ef5298/tests/admin_filters/tests.py#L1082). 

I also think a name like `assertDateHierarchy` might imply it's avaliable as part of `TestCase` (such as other Django asserts like `assertNumQueries`).